be/src/exec/pipeline/pipeline_driver.cpp

```diff
@@ -398,6 +394,19 @@ void PipelineDriver::_close_operators(RuntimeState* runtime_state) {
 }
 
 void PipelineDriver::finalize(RuntimeState* runtime_state, DriverState state) {
+    int64_t time_spent = 0;
```

Comment:
Why `finalize` will take a long time? is that another problem?

Reply:
The `finalize` phase need to do some release jobs of fragment context and query context for the last time count down.
This costs about 0.1~08s, if there is 200 fragments of a large query, then it will cost 20s.

But it is indeed too heavy, maybe we could optimize this later.